simpleviewinc/tap-events-force
# ZEN-522: session unbooking bugfix

**Ticket**: [ZEN-522](https://jira.simpleviewtools.com/browse/zen-522)

## Context

* There is a bug in the booking logic
* If **at the time of initial load**, a session has no capacity remaining, its booking button will be disabled even if there are attendees on the booking/waiting lists that the user set
* Sessions should not be disabled in this scenario so that the user is able to remove them from the booking / waiting lists

## Goal

* Fix the bug

## Updates

* `src/mocks/eventsforce/bookingStates/fullyBooked.js`
  * updated the mock data's "capacity" states so that the attendees are set to be already booked to the session, for testing this scenario
* `src/utils/models/sessions/getDisabled.js`
  * updated the `getDisabled` utility for the booking state factory to ensure that a button is **not** disabled if there are any attendees on the booking or waiting lists.

## Testing
* `keg evf pack run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-511-session-unbook`
1. http://evf-ZEN-511-session-unbook.local.kegdev.xyz?state=fullyBooked.single.stopped
  * in this state, the session is at capacity, but the attendee is not on the booking or waiting list, so it should be disabled
  * verify that the booking button is disabled
2. http://evf-ZEN-511-session-unbook.local.kegdev.xyz?state=fullyBooked.single.capacity
  * in this state, the session is at capacity, but the attendee is on the booking list this time
  * verify that the booking button is enabled, and that you can select it to unbook the attendee
    * after the attendee is unbooked, the button will say "fully booked", but that is expected: in this testing mode, the mock  session capacity doesn't get updated (that's something events force handles)
3. http://evf-ZEN-511-session-unbook.local.kegdev.xyz?state=fullyBooked.group.stopped
  * same as step 1, but now in group-booking mode
  * verify that the booking button is disabled
4. http://evf-ZEN-511-session-unbook.local.kegdev.xyz?state=fullyBooked.group.capacity
  * same as step 3, but now in group-booking mode
  * verify that the booking button is enabled and that you can:  
    * open the modal
    * unbook attendees
